runtimes/core/src/api/gateway/mod.rs

```diff
@@ -109,7 +109,7 @@ impl Gateway {
         self.inner.shared.auth.as_ref()
     }
 
-    pub async fn serve(self, listener: TcpListener) -> anyhow::Result<()> {
+    pub async fn serve(self, listen_addr: String) -> anyhow::Result<()> {
```

Comment:
Given that we pass it as a ref to `proxy.add_tcp`, can we make this take `&str`?